### ISSUE TITLE
Question: WinUI 3 desktop app UI goes to idle state when not on focus. 

### ISSUE THREAD

**Issue by heetaekim** [microsoft/microsoft-ui-xaml#7149]
Hi,

Apologies if this topic is already covered. However, I simply couldn't find similar issue/behavior anywhere.

I have an WinUI 3 desktop app which is a trading application which displays real-time live prices and volumes.

When the app is not on focus, the UI seems to go into some sort of an idle state after some time (I know because price data is not updated on the screen). Only when I click on the app screen to bring it to focus, it resumes updating the prices.

At first, I thought UI thread is being blocked by some long running methods, but after examining the code, I cannot see anything like that happening and the UI thread is not being overloaded.

This issue affects the usability of the app where the end user should be able to monitor the price data while working on something else on the screen.

After digging aroudn the web, I found you can set thread execution state of the app although I am not sure if these are the right thing to do.
https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-setthreadexecutionstate

Nevertherless, I have tried setting the flags and been testing, but I am still seeing screen going idle.
FYI, I have also developed similar app in WPF long time ago and never seen such issues.

Can someone help me what I am doing wrong?

Thank you.

hee-tae

**Comment by kmgallahan** [comment]
WinUI 3 apps don't have idle / sleep / power states.

My apps run for hours without focus and continue to update their UI as expected.

If you are using UWP / WinUI 2 then things are different.

**Comment by heetaekim** [comment]
@kmgallahan yes, thats what I expect from my app too and I am using WinUI 3. Been trying to optimize the UI as much as possible for past few months as well without any luck getting rid of this strange screen freeze issue

**Comment by MikeHillberg** [comment]
Agreed that this isn't something WinUI3 has the ability to do. Could it be that the UI thread is OK but the data source is somehow stalled? I had a similar issue to this one time and to help debug put a [ProgressBar](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.ProgressBar) on the window (with `IsIndeterminate` set to true) so that it's always animating. Not necessarily good for battery life, but a way to see if the UI thread is alive and ticking.

**Comment by heetaekim** [comment]
@MikeHillberg I have a textblock in the UI that displays the up time. It displays up to seconds level time so if UI is alive, then I would have expect it to tick. But when the problem happens, the up time text block stops ticking too. It is very strange...

**Comment by heetaekim** [comment]
@MikeHillberg I added the progress bar as you suggested just now. And noticed that it was still animating while the price changes in the data grid was not updating. So I think you might be right, that data source is somehow blocked.

Having said that, when I was hovering over buttons and tabs, they were not responding to my on-hover actions either (and yet the progress bar was moving). So I am little bit confused.

**Comment by heetaekim** [comment]
Hi guys,
I am actually really stuck with this now.
I've tried optimizing UI as much as possible and I know UI thread itself is ok. I have used different datagrid (telerik) with not much luck either maybe slightly better.

It just appears that the WinUI 3 app somehow stops rendering when there is no user interaction with the app for a period of time. This is beacuse, when I even hover over the app or click on the window, it suddenly comes back to life. If there was an issue with UI thread being busy, the user interaction wouldn't do anything.

I've been googling for some time now and no one seems to have same issue..
